scripts/skills/actives/rf_flaming_arrows_skill.nut

```diff
@@ -0,0 +1,219 @@
+	{
+		this.aimed_shot.create();
+		this.m.ID = "actives.rf_flaming_arrows";
+		this.m.Name = "Flaming Arrow";
+		this.m.Description = "An arrow with a flammable load which does additional fire damage and lights the target tile on fire, but is unwieldy to shoot.";
+		this.m.Icon = "skills/rf_flaming_arrows_skill.png";
+		this.m.IconDisabled = "skills/rf_flaming_arrows_skill_sw.png";
+		this.m.Overlay = "rf_flaming_arrows_skill";
+		this.m.ProjectileType = ::Const.ProjectileType.FlamingArrow;
+		this.m.InjuriesOnBody = ::Const.Injury.BurningAndPiercingBody;
+		this.m.InjuriesOnHead = ::Const.Injury.BurningAndPiercingHead;
+		this.m.AdditionalAccuracy = -10;
+		this.m.AdditionalHitChance = -4;
+		this.m.DirectDamageMult += 0.1; // Additional armor penetrating damage to reflect additional "fire" damage.
+	}
+
+	function getTooltip()
+	{
+		local ret = this.getRangedTooltip(this.getDefaultTooltip());
+
+		ret.push({
+			id = 10,
+			type = "text",
+			icon = "ui/icons/special.png",
+			text = ::Reforged.Mod.Tooltips.parseString("Will trigger a negative [morale check|Concept.MoraleCheck] for the character hit and all adjacent enemies")
+		});
+
+		ret.push({
+			id = 11,
+			type = "text",
+			icon = "ui/icons/special.png",
+			text = ::Reforged.Mod.Tooltips.parseString("If a target is hit, will light the tile on fire for " + ::MSU.Text.colorNegative(2) + " [rounds|Concept.Round]")
+		});
+
+		if (this.m.AmmoConsumed > 1)
+		{
+			ret.push({
+				id = 12,
+				type = "text",
+				icon = "ui/icons/warning.png",
+				text = "Consumes " + ::MSU.Text.colorRed(this.m.AmmoConsumed) + " ammo"
+			});
+		}
+
+		local ammo = this.getAmmo();
+
+		if (ammo > 0)
+		{
+			ret.push({
+				id = 8,
+				type = "text",
+				icon = "ui/icons/ammo.png",
+				text = "Has " + ::MSU.Text.colorPositive(ammo) + " arrows left"
+			});
+		}
+		else
+		{
+			ret.push({
+				id = 8,
+				type = "text",
+				icon = "ui/tooltips/warning.png",
+				text = ::MSU.Text.colorNegative("Needs a non-empty quiver of arrows equipped")
+			});
+		}
+
+		if (this.getContainer().getActor().isEngagedInMelee())
+		{
+			ret.push({
+				id = 9,
+				type = "text",
+				icon = "ui/tooltips/warning.png",
+				text = ::Reforged.Mod.Tooltips.parseString(::MSU.Text.colorNegative("Cannot be used because this character is [engaged|Concept.ZoneOfControl] in melee"))
+			});
+		}
+
+		return ret;
+	}
+
+	function isUsable()
+	{
+		return this.getAmmo() >= this.m.AmmoConsumed && this.aimed_shot.isUsable();
+	}
+
+	function onAfterUpdate( _properties )
+	{
+		this.aimed_shot.onAfterUpdate(_properties);
+		this.m.MaxRange += this.m.RangeModifier;
+	}
+
+	function onUse( _user, _targetTile )
+	{
+		// One ammo is consumed during the subsequent aimed_shot.onUse, so this manual loop consumes the additional ammo
+		// Start the loop at 1 because we want to consume 1 less ammo as 1 is consumed by aimed shot function call.
+		for (local i = 1; i < this.m.AmmoConsumed; i++)
+		{
+			this.consumeAmmo();
```

Comment:
You could hook the `consumeAmmo` function instead and call the base function once per `this.m.AmmoConsumed`
Like this
```
q.consumeAmmo = @(__original) function()
{
	for (local i = 1; i <= this.m.AmmoConsumed; ++i)
	{
		__original();
	}
}
```